src/main/kotlin/org/jetbrains/research/ml/ast/transformations/deadcode/DeadCodeRemovalHeuristicVisitor.kt

```diff
@@ -75,16 +68,19 @@ internal class DeadCodeRemovalHeuristicVisitor(private val commandPerformer: ICo
      */
     private fun handleWhileFalseStatement(node: PyWhileStatement) {
         if (node.whilePart.condition?.evaluateBoolean() == false) {
-            // Todo: replace { } with the real undo
-            commandPerformer.performCommand(
-                Command(
-                    { node.delete() },
-                    { },
-                    "Delete \"while\"-node with false condition"
-                )
-            )
+            commandPerformer.performCommand(DeleteCommand(node).getCommand("Delete \"while\"-node with false condition"))
+//            commandPerformer.performCommand(Command(runInWCA(node.project){ node.delete() }, { }, ))
         }
     }
 }
 
 private fun PyExpression.evaluateBoolean(): Boolean? = PyEvaluator.evaluateAsBoolean(this)
+
+
+fun <T>runInWCA(project: Project, action: () -> T): Callable<T> {
```

Comment:
Should we move it into utils?

Reply:
Our utils require serious refactoring, so i made a new class for this function. it's temporal though, so after implementing all commands we can remove it